pkg/client/typed/dynamic/client.go

```diff
@@ -91,6 +98,13 @@ type ResourceClient struct {
 	ns       string
 }
 
+// SubresourceClient is an API interface to a specific subresource of a resource under a
+// dynamic client.
+type SubresourceClient struct {
+	*ResourceClient
```

Comment:
Although I don't follow @deads2k argument exactly. The name that's passed into Get is the name of the resource on which we're going to get the subresource from.